lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js

```diff
@@ -8,47 +8,74 @@
  * @fileoverview  A runner that launches Chrome and executes Lighthouse via a
  * bundle to test that bundling has produced correct and runnable code.
  * Currently uses `lighthouse-dt-bundle.js`.
+ * Runs in a worker to avoid messing up marky's global state.
  */
 
 import fs from 'fs';
+import os from 'os';
+import {Worker, isMainThread, parentPort} from 'worker_threads';
 
 import puppeteer from 'puppeteer-core';
 import ChromeLauncher from 'chrome-launcher';
 
 import ChromeProtocol from '../../../../lighthouse-core/gather/connections/cri.js';
 import {LH_ROOT} from '../../../../root.js';
+import {loadArtifacts, saveArtifacts} from '../../../../lighthouse-core/lib/asset-saver.js';
 
-const originalBuffer = global.Buffer;
-const originalRequire = global.require;
-if (typeof globalThis === 'undefined') {
-  // @ts-expect-error - exposing for loading of dt-bundle.
-  global.globalThis = global;
+if (!isMainThread && parentPort) {
```

Comment:
@brendankenny

Reply:
FWIW, since the worker only does one thing and then dies, it doesn't actually need to receive messages. The worker can just [load from `workerData` ](https://nodejs.org/api/worker_threads.html#workerworkerdata) and get rid of some of this complexity.